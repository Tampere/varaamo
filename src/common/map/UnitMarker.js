import L from 'leaflet';
import icon from 'leaflet/dist/images/marker-icon-2x.png';
import PropTypes from 'prop-types';
import React from 'react';
import get from 'lodash/get';
import { Marker } from 'react-leaflet';

import highlightedIcon from '../../../app/assets/map/highlighted-marker-icon-2x.png';

const getIcon = (count, isHighlighted = false) => {
  return L.divIcon({
    className: 'marker-icon',
    iconAnchor: [12, 48],
    iconSize: [25, 48],
    html: `<div>${count}</div><img src="${isHighlighted ? highlightedIcon : icon}" />`,
  });
};

const UnitMarker = ({
  unit,
  resources,
  isHighlighted,
  onClick,
}) => {
  const coordinates = get(unit, 'location.coordinates');

  return (
    <Marker
      icon={getIcon(resources.length, isHighlighted)}
      onClick={() => {
        if (onClick) {
          onClick(unit, resources);
        }
      }}
      position={[coordinates[1], coordinates[0]]}
    />
  );
};

UnitMarker.propTypes = {
  unit: PropTypes.object.isRequired,
  resources: PropTypes.array,
  onClick: PropTypes.func,
  isHighlighted: PropTypes.bool,
};

export default UnitMarker;
