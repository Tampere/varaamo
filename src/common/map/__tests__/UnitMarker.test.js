import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import UnitMarker from '../UnitMarker';

describe('UnitMarker', () => {
  const unit = {
    id: 'testUnit',
    location: {
      coordinates: [24.975634, 60.20843],
    },
  };

  test('renders correctly', () => {
    const wrapper = shallow(
      <UnitMarker
        resources={[{ id: 'foo' }]}
        unit={unit}
      />,
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test('show number of resources when more than 1 resources', () => {
    const wrapper = shallow(
      <UnitMarker
        resources={[
          { id: 'foo' },
          { id: 'bar' },
        ]}
        unit={unit}
      />,
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test('onClick function works', () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <UnitMarker
        onClick={onClick}
        resources={[
          { id: 'foo' },
        ]}
        unit={unit}
      />,
    );

    wrapper
      .simulate('click');

    expect(onClick.mock.calls.length).toBe(1);
    expect(onClick.mock.calls[0][0].id).toBe('testUnit');
    expect(onClick.mock.calls[0][1][0].id).toBe('foo');
  });
});
