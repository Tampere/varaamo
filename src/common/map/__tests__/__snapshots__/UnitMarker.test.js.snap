// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnitMarker renders correctly 1`] = `
<ForwardRef(Leaflet(Marker))
  icon={
    NewClass {
      "_initHooksCalled": true,
      "options": Object {
        "className": "marker-icon",
        "html": "<div>1</div><img src=\\"test-file-stub\\" />",
        "iconAnchor": Array [
          12,
          48,
        ],
        "iconSize": Array [
          25,
          48,
        ],
      },
    }
  }
  onClick={[Function]}
  position={
    Array [
      60.20843,
      24.975634,
    ]
  }
/>
`;

exports[`UnitMarker show number of resources when more than 1 resources 1`] = `
<ForwardRef(Leaflet(Marker))
  icon={
    NewClass {
      "_initHooksCalled": true,
      "options": Object {
        "className": "marker-icon",
        "html": "<div>2</div><img src=\\"test-file-stub\\" />",
        "iconAnchor": Array [
          12,
          48,
        ],
        "iconSize": Array [
          25,
          48,
        ],
      },
    }
  }
  onClick={[Function]}
  position={
    Array [
      60.20843,
      24.975634,
    ]
  }
/>
`;
